### ISSUE TITLE
Feature request

### ISSUE THREAD

**Issue by mavidser** [localsend/localsend#2023]
### Is your feature request related to a problem? Please describe.

No. Just a clear Usability upgrade.

### Describe the solution you'd like

Add as a macOS extension so LocalSend can be visible in the 'Share' context menu in finder. This can open the device picker in Localsend.

### Describe alternatives you've considered

The current method of opening the app and sending the file.
I can look into building this feature - but want to check the opinion on maintainers

**Comment by Tienisto** [comment]
Do you use the latest version?

**Comment by mavidser** [comment]
Ah, I updated to latest, and a share target is there. Thanks!

Though, LocalSend doesn't remember the file it launched for, and starts in the default Receive tab, when opened from Share menu.

**Comment by j-lakeman** [comment]
I can confirm that. Thoughts @Tienisto?

**Comment by Tienisto** [comment]
It does remember the shared file on my side.

Maybe related: Can you drop a file into the menu bar icon?

**Comment by j-lakeman** [comment]
You mean drop it into the LocalSend window?

**Comment by ShlomoCode** [comment]
@Tienisto I can confirm that this happens to me with the released app, while the development build works. Dragging to the menu bar icon still works.
After a little investigation, it seems to be related to a shared group not working properly, the `ShareExtention` stores the new file in:
```
/Users/shlomo/Library/Containers/org.localsend.localsendApp.ShareExtension/Data/Library/Preferences/3W7H4PYMCV.localsend.shared_group.plist
```
While the main app accesses:
```
/Users/shlomo/Library/Containers/org.localsend.localsendApp/Data/Library/Preferences/3W7H4PYMCV.localsend.shared_group.plist
```

**Comment by ShlomoCode** [comment]
I see that dragging to the Dock icon doesn't work either, so maybe it's not related to the shared group 🤔 
@Tienisto Does dragging to the Dock icon works for you?

**Comment by Tienisto** [comment]
@ShlomoCode No, dragging a file into the Dock icon somehow doesn't work (anymore?) 💀

**Comment by 70sh1** [comment]
May I ask if a fix for this issue is planned? I would really appreciate this feature.

**Comment by ShlomoCode** [comment]
> [@ShlomoCode](https://github.com/ShlomoCode) No, dragging a file into the Dock icon somehow doesn't work (anymore?) 💀

I got it, [`application:openFile`](https://developer.apple.com/documentation/appkit/nsapplicationdelegate/application(_:openfile:)) and [`application:openFiles`](https://developer.apple.com/documentation/appkit/nsapplicationdelegate/application(_:openfiles:)) are not called at all!
It turns out that macOS calls [`application:openURLs`](https://developer.apple.com/documentation/appkit/nsapplicationdelegate/application(_:open:)) when dragging files to the Dock icon... I don't understand how it worked in the past... I'll will PR soon

And it also fixed the share extension!

**Comment by 70sh1** [comment]
After additional testing it turned out to be an issue with `brew` installed version of localsend. However, the App Store version works fine. Tested on two MacBook Air 13' M4 's.

**Comment by ShlomoCode** [comment]
> After additional testing it turned out to be an issue with `brew` installed version of localsend. However, the App Store version works fine. Tested on two MacBook Air 13' M4 's.

Hmm, that's very strange! Maybe it's related to the Hardened Runtime. In the current version, the App Store build is not hardened.
Anyway, dragging a file to the Dock icon didn't work in the Mac App Store version either, but it's fixed now.

**Comment by vdmkotai** [comment]
@ShlomoCode sorry for bothering. Can you take a look at my PR, please?

**Comment by ShlomoCode** [comment]
I'll look
